download_and_create_reference_datasets/v02/mito/write_combined_mito_reference_data_hts.py

```diff
@@ -0,0 +1,191 @@
+        'input_path': 'gs://seqr-reference-data/GRCh38/mitochondrial/MITOMAP/Mitomap Confirmed Mutations Feb. 04 2022.tsv',
+        'input_type': 'tsv',
+        'output_path': 'gs://seqr-reference-data/GRCh38/mitochondrial/MITOMAP/Mitomap Confirmed Mutations Feb. 04 2022.ht',
+        'annotate': {
+            'locus': lambda ht: hl.locus('chrM', hl.parse_int32(ht.Allele.first_match_in('m.([0-9]+)')[0])),
+            'alleles': lambda ht: ht.Allele.first_match_in('m.[0-9]+([ATGC]+)>([ATGC]+)'),
+            'pathogenic': lambda ht: hl.is_defined(ht['Associated Diseases'])
+        },
+    },
+    'mitimpact': {
+        'input_path': 'https://mitimpact.css-mendel.it/cdn/MitImpact_db_3.0.7.txt.zip',
+        'input_type': 'tsv',
+        'output_path': 'gs://seqr-reference-data/GRCh38/mitochondrial/MitImpact/MitImpact_db_3.0.7.ht',
+        'annotate': {
+            'locus': lambda ht: hl.locus('chrM', hl.parse_int32(ht.Start)),
+            'alleles': lambda ht: [ht.Ref, ht.Alt],
+            'APOGEE_score': lambda ht: hl.parse_float(ht.APOGEE_score),
+        },
+    },
+    'hmtvar': {
+        # The data source has certificate expiration issue. Manually downloaded from https://www.hmtvar.uniba.it/api/main/
+        'input_path': 'gs://seqr-reference-data/GRCh38/mitochondrial/HmtVar/HmtVar%20Jan.%2010%202022.json',
+        'input_type': 'json',
+        'output_path': 'gs://seqr-reference-data/GRCh38/mitochondrial/HmtVar/HmtVar%20Jan.%2010%202022.ht',
+        'annotate': {
+            'locus': lambda ht: hl.locus('chrM', hl.parse_int32(ht.nt_start)),
+            'alleles': lambda ht: [ht.ref_rCRS, ht.alt],
+            'disease_score': lambda ht: hl.parse_float(ht.disease_score),
+        },
+    },
+    'helix': {
+        'input_path': 'https://helix-research-public.s3.amazonaws.com/mito/HelixMTdb_20200327.tsv',
+        'input_type': 'tsv',
+        'output_path': 'gs://seqr-reference-data/GRCh38/mitochondrial/Helix/HelixMTdb_20200327.ht',
+        'field_types': {'counts_hom': hl.tint32, 'AF_hom': hl.tfloat64, 'counts_het': hl.tint32,
+                        'AF_het': hl.tfloat64, 'max_ARF': hl.tfloat64, 'alleles': hl.tarray(hl.tstr)},
+        'annotate': {
+            'locus': lambda ht: hl.locus('chrM', hl.parse_int32(ht.locus.split(':')[1])),
+        },
+    },
+    'clinvar': {
+        'input_path': 'ftp://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/clinvar.vcf.gz',
+        'input_type': 'vcf',
+        'output_path': 'gs://seqr-reference-data/GRCh38/mitochondrial/clinvar/clinvar.GRCh38.chrM.ht',
+        'annotate': {
+            'ALLELEID': lambda ht: ht.info.ALLELEID,
+            'CLNSIG': lambda ht: ht.info.CLNSIG,
+            'CLNREVSTAT': lambda ht: ht.info.CLNREVSTAT,
+        },
+    },
+    'dbnsfp': {
+        'input_path': 'gs://seqr-reference-data/GRCh38/all_reference_data/v2/combined_reference_data_grch38-2.0.4.ht',
```

Comment:
I don't really want to pay to store a subset of the exact same data twice. There is no good reason you can't tweak the config for `dbnsfp_mito` to use the full dbnsfp ht as its input

Reply:
It is possible only if we add a `filter` configuration and we update the `joined_ht()` function to filter out the variants based on the configuration.

Reply:
sure that sounds fine

Reply:
No. We really want to keep single source files for the reference data, the exact same way we already do for all the other reference data. In the future if we need to update one reference source and keep the other the same we want to have all the other reference sources saved from the last time so we don't need to recompute them. Again, if you are having trouble understanding the approach we are taking here, I would ask you to schedule time to meet and discuss

Reply:
As much as possible we want this to maintain the exact same behavior we already have for reference data, and this whole pipeline in general. In general the answer to "should I do this differently" is always no. The small amount of possible savings will never be worth the technical and mental overhead of maintaining two different ways of doing what is basically the same thing. The mito pipeline NEEDS to run the same way a the regular SNP pipeline wherever possible. If you think there are better ways to do things, make a ticket in the backlog and when we have bandwidth in the future we can make improvements to both the main pipeline and the mito pipeline through shared code. But for now, unless there is an actual data reason things need to be different, they need to be the same